BUILDING.md

```diff
@@ -0,0 +1,249 @@
+   
+7. You can now run and debug Audacity!
+      
+Generally, steps 1-5 are only needed for the first-time configuration. Then, after you've generated the solution - you can open it in Visual Studio next time. In case when the project configuration has changed - the IDE will invoke CMake internally. 
+
+> Conan Center provides prebuilt binaries only for **x64**. Configuring the project for Win32 will take much longer, as all the 3d party libraries will be built during the configuration.
+
+## macOS
+
+1. Clone Audacity from the Audacity GitHub project. 
+  
+    ```
+    $ git clone https://github.com/audacity/audacity/
+    ```
+
+2. Configure Audacity using CMake:
+   ```
+   $ mkdir build && cd build
+   $ cmake -GXCode -T buildsystem=1 ../audacity
+   ```
+
+3. Open Audacity XCode project:
+   ```
+   $ open Audacity.xcodeproj
+   ```
+   and build Audacity using the IDE. 
+
+Steps 1 and 2 are only required for the first-time builds. 
+
+Alternatively, you can use **CLion**. If you chose so, open the directory where you have cloned Audacity using CLion and you are good to go.
+
+At the moment, we only support **x86_64** builds. It is possible to build using AppleSilicon harware, but **mad** and **id3tag** should be disabled:
+
+```
+cmake -GXCode -T buildsystem=1 -Daudacity_use_mad="off" -Daudacity_use_id3tag=off ../audacity
+```
+
+## Linux & Other OS
+
+1. Clone Audacity from the Audacity GitHub project. 
+  
+    ```
+    $ git clone https://github.com/audacity/audacity/
+    ```
+
+2. Configure Audacity using CMake:
+   ```
+   $ mkdir build && cd build
+   $ cmake -G "Unix Makefiles" -Daudacity_use_ffmpeg=loaded ../audacity
+   ```
+
+3. Build Audacity:
+   ```
+   $ make -j'nproc`
+   ```
+
+4. Testing the build:
+   Adding a "Portable Settings" folder allows Audacity to ignore the settings of any existing Audacity installation.
+   ```
+   $ cd bin/Release
+   $ mkdir "Portable Settings"
+   $ ./audacity
+   ```
+
+5. Installing Audacity
+   ```
+   $ cd <build directory>
+   $ sudo make install
+   ```
+
+## Advanced
+
+### CMake options
+
+You can use `cmake -LH` to get a list of the options available (or use CMake GUI or `ccmake`).
+
+The most notable options are:
+
+| Name                              | Type   | Default    | Description                                                     |
+| :-------------------------------- | :----- | :--------- | :-------------------------------------------------------------- |
+| CMAKE_BUILD_TYPE                  | STRING | Debug      | Type of the build: Debug, Release, RelWithDebInfo, MinSizeRel   |
+| CMAKE_INSTALL_PREFIX              | PATH   | /usr/local | Install path prefix, prepended onto install directories.        |
+| audacity_lib_preference           | STRING | local      | Library preference [system (if available), local]               |
+| audacity_obey_system_dependencies | BOOL   | Off        | Use only system packages to satisfy dependencies                |
```

Comment:
This table is going to be a pain to maintain in Markdown. I recommend linking to the relevant CMakeLists.txt where the options are defined, or if they are in separate files then just tell people to use the `cmake -LH` option.

If you really want a separate list, I recommend putting it in an online spreadsheet and linking to it from here. I'll PM you with more details about this.

Reply:
Yeah, that why I have called it "most notable" LOL

I guess I will move it in a separate doc a rewrite it a bit. Also, VS code makes such tables really easy to format :-)